lib/index.js

```diff
@@ -319,3 +333,134 @@ exports.socket = createSocket;
 
 exports.createDevice = createDevice;
 exports.device = createDevice;
+
+/* convenience methods to set and to get sockopts, a.k.a. the sockopt API */
+var sol = {
+  linger          : nn.NN_LINGER,
+  sndbuf          : nn.NN_SNDBUF,
+  rcvbuf          : nn.NN_RCVBUF,
+  sndtimeo        : nn.NN_SNDTIMEO,
+  rcvtimeo        : nn.NN_RCVTIMEO,
+  reconn          : nn.NN_RECONNECT_IVL,
+  maxreconn       : nn.NN_RECONNECT_IVL_MAX,
+  sndprio         : nn.NN_SNDPRIO,
+  tcpnodelay      : nn.NN_TCP_NODELAY
+}
+
+function linger(number){
+  if(number){
```

Comment:
should be fine, but i need to move the nn_socket() call out and up into https://github.com/nickdesaulniers/node-nanomsg/blob/master/lib/index.js#L291-L294 and then just give it to the binding as a parameter to 
```js
new Socket(domain, type);
```